opensearch_py_ml/ml_commons/ml_commons_client.py

```diff
@@ -359,6 +360,34 @@ def _get_task_info(self, task_id: str):
             url=API_URL,
         )
 
+    def search_task(self, json_input) -> object:
+        """
+        This method searches a task from opensearch cluster (using ml commons api)
+        :param json: json input for the search request
+        :type json: string or dict
+        :return: returns a json object, with detailed information about the searched task
+        :rtype: object
+        """
+
+        API_URL = f"{ML_BASE_URI}/tasks/_search"
+
+        if isinstance(json_input, str):
+            try:
+                json_obj = json.loads(json_input)
+                API_BODY = json.dumps(json_obj)
+            except json.JSONDecodeError:
+                return "Invalid JSON string passed as argument."
+        elif isinstance(json_input, dict):
+            API_BODY = json.dumps(json_input)
+        else:
+            return "Invalid JSON object passed as argument."
```

Comment:
Can we cover this line with testing?

Reply:
"return "Invalid JSON object passed as argument." this block

Reply:
Ah I see maybe you mean try catch block missed for dict case. I will add it.

Reply:
> "return "Invalid JSON object passed as argument." this block

Ok